### PR TITLE
Closes #6603 use the created time when creating a download notification

### LINKED ISSUE #6603: Re-use the created time when creating a download notification.
We experienced some issues in [certain specific devices](https://github.com/mozilla-mobile/fenix/issues/8930#issuecomment-610900304), where the OS usages the [when](https://developer.android.com/reference/android/app/Notification#when) value to sort the notifications in the status bar, when we are downloading a notification, [we are constantly creating a new Notifications object](https://github.com/mozilla-mobile/android-components/blob/dfd529b1cc8ef246d9a8504fe98c67f8a2576e76/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadNotification.kt#L54) and subsequently updating the [when](https://developer.android.com/reference/android/app/Notification#when) value, this causes that every time we update the progress of the notification the some OSs re-order the notifications, causing issue like the one below:

 OnePlus A3 (Android 6.0.1)

### PR DESCRIPTION
**Pull request by Amejia481**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadNotification.kt [hunk 1]
```diff
@@ -68,16 +71,19 @@ internal object DownloadNotification {
...
 
+        val downloadState = downloadJobState.state
         return NotificationCompat.Builder(context, channelId)
             .setSmallIcon(R.drawable.mozac_feature_download_ic_download)
             .setContentTitle(downloadState.fileName)
             .setContentText(context.getString(R.string.mozac_feature_downloads_paused_notification_text))
             .setColor(ContextCompat.getColor(context, R.color.mozac_feature_downloads_notification))
             .setCategory(NotificationCompat.CATEGORY_PROGRESS)
             .setOngoing(true)
+            .setWhen(downloadJobState.createdTime)
```
Comment (csadilek) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadNotification.kt:85]: I am wondering if this should be a separate timestamp i.e. the time the notification happened ( == the time the download was paused, not created)?

As this is a different notification than the in progress one.

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadNotification.kt [hunk 2]
```diff
@@ -86,12 +92,15 @@ internal object DownloadNotification {
...
      */
-    fun createDownloadCompletedNotification(context: Context, downloadState: DownloadState): Notification {
+    fun createDownloadCompletedNotification(context: Context, downloadJobState: DownloadJobState): Notification {
         val channelId = ensureChannelExists(context)
+        val downloadState = downloadJobState.state
 
         return NotificationCompat.Builder(context, channelId)
             .setSmallIcon(R.drawable.mozac_feature_download_ic_download_complete)
             .setContentTitle(downloadState.fileName)
+            .setWhen(downloadJobState.createdTime)
```
Comment (csadilek) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadNotification.kt:102]: I am wondering if this should be a separate timestamp i.e. the time the notification happened ( == the time the download was completed, not created)?

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadNotification.kt (hunk 1 above)
Comment (Amejia481) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadNotification.kt:85]: It really depends on how we want the UI to react, if we use a different timestamp, the notification will change positions when we click the resume button. If we use the same the notification will remain in the same position. I'm not sure what should be the correct behaviour?

Comment (Amejia481) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadNotification.kt:85]: 👍

**Comment by Amejia481** [comment]
bors r=csadilek

**Pull request merged**